pandas/tests/groupby/test_categorical.py

```diff
@@ -963,3 +964,72 @@ def test_shift(fill_value):
                               categories=['a', 'b', 'c', 'd'], ordered=False)
     res = ct.shift(1, fill_value=fill_value)
     assert_equal(res, expected)
+
+
+@pytest.fixture
+def df_cat(df):
+    df_cat = df.copy()[:4]  # leave out some groups
+    df_cat['A'] = df_cat['A'].astype('category')
+    df_cat['B'] = df_cat['B'].astype('category')
+    df_cat['C'] = pd.Series([1, 2, 3, 4])
+    yield df_cat
+
+
+@pytest.mark.parametrize('operation, index', [
+    ('agg', MultiIndex.from_frame(
+        pd.DataFrame({'A': ['foo', 'foo', 'bar', 'bar'],
+                      'B': ['one', 'two', 'one', 'three']
+                      }, dtype='category'))),
+    ('apply', MultiIndex.from_frame(
+        pd.DataFrame({'A': ['foo', 'foo', 'bar', 'bar'],
+                      'B': ['one', 'two', 'one', 'three']
+                      })))])
+def test_seriesgroupby_observed_true(df_cat, operation, index):
+    # GH 24880
+    expected = pd.Series(data=[1, 3, 2, 4], index=index, name='C')
```

Comment:
Series and DataFrame are already imported, so you could remove the pd prefixes, here and elsewhere in the tests.